packages/collector/src/announceCycle/agentready.js

```diff
@@ -142,3 +146,39 @@ function fireMonitoringEvent() {
     }
   });
 }
+
+function sendEOLEvent() {
+  agentConnection.sendEvent(
+    {
+      title: `Node.js version ${process.versions.node} reached its end of life`,
+      text:
+        'This version no longer receives updates or security fixes and might contain unfixed vulnerabilities.\n\n' +
+        'Please consider upgrading Node.js to an active version.\n\n' +
+        'For a list of active versions visit ' +
+        '[https://nodejs.org/en/about/releases/](https://nodejs.org/en/about/releases/)',
+      plugin: 'com.instana.forge.infrastructure.runtime.nodejs.NodeJsRuntimePlatform',
+      id: pidStore && typeof pidStore.getEntityId === 'function' ? pidStore.getEntityId() : undefined,
+      timestamp: Date.now(),
+      duration: ELEVEN_MINUTES,
+      severity: 5
```

Comment:
Exactly, I opted by using the type only. If you provide a value not covered by `ProblemSeverity`, `tsc` will complain.
BUT, to make it even more clear, you are right. We can have some constants defined somewhere.
I am not sure about the enum, to be fair, because I don't think we use such thing very often, or at all, maybe?
The magic number issue is technically gone, IF you use some fancy IDE or build the code (TS screaming at you). But by just batting an eye to it, it really looks ugly.
Let me review this and see if an enum fits well. Otherwise, I will just create constants somewhere (not sure where yet, as the collector package has no constants.js)

Reply:
I really have no strong opinion here. As I said, from my perspective it is okay as it is. Constants that are centrally defined could also be very much good enough if you want to make it a bit more self-explaining.